### DISCUSSION TITLE
How do I call the 'sendMessage' function OUTSIDE of a react component?

### DISCUSSION THREAD

**Discussion by LUK3ARK** [jeffreylanters/react-unity-webgl#448] [Help Wanted!]
Hello, I am currently trying to create a standalone function that when called by unity, automatically returns a value via the sendMessage function. I have been able to create this inside of a component, however I can only call a function from the unity side.

I might be overlooking something really easy, but how can I expose the sendMessage function outside of the component and into pure Javascript? Bear in mind that this function is provided by a hook.

**Comment by devilofdev** [comment]
I couldn't do sendMessage on the component, so I got it from the component through global state management like Redux and sentMessage on the page with unityprovider.

**Reply by LUK3ARK** [reply]
I was able to do by making a reference to the sendMessage function outside of my component definition

I had issue when unity was not instantiated, so make sure to put it in a useEffect function, when the unityContext is available, then set the reference
